Rewrote this repository in Rust

SOF3 opened the pull request:
This would compile README into an executable rust file that would output the same contents as the markdown content.

One can obtain the executable by running `just build`. Currently the justfile only supports Linux-like systems.

SOF3 committed:
Rewrote this repository in Rust

SOF3 commented on README.md:
```diff
@@ -18,7 +21,14 @@ No. This is a joke.
 
 ### Y U HATE RUST SO MUCH?
 
-I don't, actually. I believe that those who spend their time asking people to rewrite their projects are probably not themselves active Rust developers, as those active devs are probably busy [writing memory-safe code](https://trac.torproject.org/projects/tor/ticket/11331).
```
HTTPS links have `//` which prevents the `()` from being paired, causing errors at the lexing stage. They have been migrated to footnote style which does not wrap URLs inside brackets.

SOF3 commented on riir-macro/src/lib.rs:
```diff
@@ -0,0 +1,142 @@
+fn test() {
+    use std::{fs, path::PathBuf};
+
+    let path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("test.txt");
```
The test file was made external instead of using `quote!()`, because `quote!()` does not allow literal expressions with a `#`.

SOF3 commented:
The CI is currently failing because of rust-lang/cargo#8354 failing to install `just`. It has been tested to work on older nightly versions.

ansuz commented:
This is an impressive display of over-engineering. Kudos!

The only thing that seems to be missing at this point is instructions for how to build the readme within the readme.

SOF3 committed:
Updated building instructions

SOF3 commented:
I have added the instructions to build the readme. I have also added a `rust-toolchain` file so that a nightly compiler with `proc_macro_span` support is used.

ansuz commented on README.md:
```diff
@@ -18,7 +21,26 @@ No. This is a joke.
+This readme is written in Rust to further rebut arguments that Rust has low readability.
+The readme can be built into an executable by running the [just][just] command:
+
+> just build
```
code blocks can be inserted in markdown by indenting with four spaces.

ansuz commented on README.md:
```diff
@@ -18,7 +21,26 @@ No. This is a joke.
 ![](rust.png)
+
+## Building
+This readme is written in Rust to further rebut arguments that Rust has low readability.
```
Who is arguing this? I'd prefer you not use my repo to pick fights with even more random people on the internet.

SOF3 replied:
k, could delete this

SOF3 committed:
Fixed formatting

emerson-proenca commented:
Consider adding a [unsafe] block when referecing "### Y U HATE RUST SO MUCH?"
